src/core/main.c

```diff
@@ -1865,6 +1867,14 @@ static int initialize_runtime(
                 }
         }
 
+        if (arg_system && arg_no_new_privs) {
+                r = prctl(PR_SET_NO_NEW_PRIVS, 1, 0, 0, 0);
+                if (r < 0) {
+                        *ret_error_message = "Failed to disable new privileges";
+                        return log_emergency_errno(r, "Failed to disable new privileges: %m");
```

Comment:
prctl() returns its error in `errno`, not `r`. Hence please change the first parameter of `log_emergency_errno()` to `errno`.

Reply:
Ah, I incorrectly assumed I can follow the same approach as for the capability bounding set.